docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,538 @@
+            The compatibility MUST be referenced by `id`.
+
+          - **`to`** *object*
+
+            This REQUIRED property references a target compatibility defined in the `spec.compatibilities.id` field.
+            Only one property is allowed _all_, _oneOf_ or _noneOf_.
+
+            - **`all`** *array of strings*
+
+              This OPTIONAL property references a list of target compatibility nodes defined in the `spec.compatibilities` with a condition that all nodes have to pass that container is considered to be compatible with a host.
+              The compatibility MUST be referenced by `id`.
+
+            - **`oneOf`** *array of strings*
+
+              This OPTIONAL property references a list of target compatibility nodes defined in the `spec.compatibilities` with a condition that at least one node has to pass that container is considered to be compatible with a host.
+              The compatibility MUST be referenced by `id`.
+
+            - **`noneOf`** *array of strings*
+
+              This OPTIONAL property references a list of target compatibility nodes defined in the `spec.compatibilities` with a condition that any node on the list cannot pass that container is considered to be compatible with a host.
+              The compatibility MUST be referenced by `id`.
+
+      - **`validationCriteria`** *array of objects*
+
+        This REQUIRED property defines compatibility criteria.
+
+        - **`graphs`** *array of strings*
+
+          This REQUIRED property references graphs names from `spec.relations.graphs`. The graphs are used to validate compatibility.
+
+        - **`condition`** *string*
+
+          This REQUIRED property specify the compatibility validation rule.
+          Allowed values: _all_ - all referenced graphs must pass; _one_ - one referenced graph must pass from the list.
+
+        - **`annotations`** *string-string map*
+
+          This OPTIONAL property describes user defined annotations.
+
+Example of use:
+
+- Simple requirements - container supporting only NVIDIA GPUs
+
+```json
+{
+  "spec": {
+    "compatibilities": [
+      {
+        "id": "nvidiaGPU",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "hardware.pci.class-id": "0380",
+          "hardware.pci.vendor-id": "10de",
+          "kernel.configuration": [
+            "CONFIG_HOTPLUG_PCI_PCIE=y",
+            "CONFIG_MODULES=y",
+            "CONFIG_MODULE_UNLOAD=y",
+            "CONFIG_PCI_MMCONFIG=m",
+            "CONFIG_DRM_NOUVEAU=n"
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+
+- Complex requirements with relations - container supporting multiple GPUs
+
+```json
+{
+  "spec": {
+    "compatibilities": [
+      {
+        "id": "intelCpu",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "hardware.cpu.vendor": "GenuineIntel",
+          "hardware.cpu.virtualization": "VT-x",
+          "kernel.cmdline": ["intel_iommu=on"]
```

Comment:
This is problematic to have different formats to be parsed depending on the field. E.g., think of structs in Go - you'd have to map it to an interface and then check for each type. Ideally everything is just a string.

Reply:
Indeed it would have to be an interface converted to appropriate type. A plugin (or whatever thing) would have to implement method with type assertion.

Let me explore this, maybe having domain format with string value would be good enough from user experience point of view. However, still you would have to parse it that the appropriate plugin can get the appropriate input.

Reply:
I didn't find any problems with the format you propose so I adjusted the proposal accordingly.